parse_lib.py

```diff
@@ -95,8 +94,9 @@ def resolve_relative_resource_urls(html_string):
     imgs = html.find_all("img", src=True)
     equations = html.find_all("object", data=True)
     list(map(update_anchor_href, anchors))
```

Comment:
I know this wasn't touched in this PR, but I noticed it so I will comment.

We aren't really "mapping" anything from one iterable to another, so I think it is misleading to have side-effects inside of a `map` call.

Because we depend on the side-effects, I think it is better to be imperative and not functional.  This avoids these misleading semantics related to `map`.  E.g.:

```
for a in anchors:
    update_anchor_href(a)
```